### PR TITLE
Sub: Set time with GCS

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

**Comment by patrickelectric** [comment]
@jaxxzer ^

**Comment by rmackay9** [comment]
This change fits in OK with @peterbarker's PR? https://github.com/ArduPilot/ardupilot/pull/8121

Just want to make sure we're not tripping over each other.

**Comment by jaxxzer** [comment]
The base branch of this is Sub-3.5. This is to get the same functionality into our stable release, with a diff that I am more comfortable with and has no conflicts.

This patch will not go into master.

**Comment by jaxxzer** [comment]
I hope this does not step on any toes or hurt any nerves on the dev team. If there are any objections or concerns to this behavior/workflow, please let me know.

**Comment by rmackay9** [comment]
@patrickelectric, no problems on my side, you guys are the sub maintainers after all.  I actually hadn't noticed that it wasn't the master branch.

File: libraries/AP_HAL_Linux/Util.cpp [hunk 1]
```diff
@@ -87,21 +87,23 @@ void Util::_toneAlarm_timer_tick() {
...
 {
 #if CONFIG_HAL_BOARD_SUBTYPE != HAL_BOARD_SUBTYPE_LINUX_NONE
+    _system_time_was_set = true;
+
     timespec ts;
-    ts.tv_sec = time_utc_usec/1.0e6;
-    ts.tv_nsec = (time_utc_usec % 1000000) * 1000;
-    clock_settime(CLOCK_REALTIME, &ts);    
-#endif    
+    ts.tv_sec = time_utc_usec/1000000ULL;
```
Comment (jaxxzer) [inline libraries/AP_HAL_Linux/Util.cpp]: why?

File: libraries/GCS_MAVLink/GCS.h [hunk 2]
```diff
@@ -164,7 +164,7 @@ class GCS_MAVLINK
     void send_accelcal_vehicle_position(uint8_t position);
 
     // return a bitmap of active channels. Used by libraries to loop
-    // over active channels to send to all active channels    
+    // over active channels to send to all active channels
     static uint8_t active_channel_mask(void) { return mavlink_active; }
```
Comment (jaxxzer) [inline libraries/GCS_MAVLink/GCS.h]: this is all whitespace? take it out of the patch please.

**Comment by patrickelectric** [comment]
@jaxxzer done.

**Comment by jaxxzer** [comment]
perfect, thanks

**Pull request merged**